Segmentation fault: 11 / Abort trap: 6

This was reported on MacPorts bug tracker: https://trac.macports.org/ticket/55236
- version: 20171027

I'm quoting the relevant part of the original report plus the crash dumps below.

~I was not able to reproduce the issue on my environment.~ Scratch that, I can also reproduce the same crashes on macOS 10.13 using the exact config files reported. This is the (shortened) version info for **my** neomutt (**not** from the original reporter):
```
NeoMutt 20171027

System: Darwin 17.2.0 (x86_64)
ncurses: ncurses 6.0.20170916 (compiled with 6.0)
libidn: 1.33 (compiled with 1.33)

Compiler:
unknown compiler

Configure options: --prefix=/opt/local --with-idn=/opt/local --with-ncurses=/opt/local --with-nls=/opt/local --with-ssl=/opt/local

Compilation CFLAGS: -pipe -Os -arch x86_64 -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/opt/local/include

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  +bkgdset +color +curs_set -debug +fcntl -flock -fmemopen +futimens 
  +getaddrinfo -gnutls -gpgme -gss -hcache -homespool +idn -locales_hack -lua 
  +meta -mixmaster +nls -notmuch +openssl +pgp +resizeterm -sasl +smime 
  +start_color +sun_attachment +typeahead 
MAILPATH="/var/mail"
PKGDATADIR="/opt/local/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/opt/local/etc"
```

Original report
---

Now, connecting to some server results in a "Segmentation fault: 11" and to other server in an "Abort trap: 6". I cannot user mutt anymore. The last good version that I can still use is @20170912.

I tested with these minimal configurations:

1) the "Segmentation fault: 11"-version
```
set imap_user="******"
set imap_pass="******"
set folder="{mail.ethz.ch/ssl}"
set spoolfile="{mail.ethz.ch/ssl}INBOX"
```
2) the "Abort trap: 6"-version:
```
set imap_user="******"
set imap_pass="******"
set folder="{imap.mail.hostpoint.ch/ssl}"
set spoolfile="{imap.mail.hostpoint.ch/ssl}INBOX"
```
the full error message is
```
mutt(85451,0x7fffc69923c0) malloc: *** error for object 0x1064a80cb: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
Abort trap: 6
```

Crash dumps (from original reporter)
---
crash dump 1:
```
Process:               neomutt [60704]
Path:                  /opt/local2/*/neomutt
Identifier:            neomutt
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        bash [60528]
Responsible:           neomutt [60704]
User ID:               25161

Date/Time:             2017-11-03 22:35:57.063 +0100
OS Version:            Mac OS X 10.12.6 (16G1036)
Report Version:        12
Anonymous UUID:        69AD0AE9-9689-39A0-1F78-564B65702D8B

Sleep/Wake UUID:       44902E2C-5BF8-4A27-85C5-2C59258C56E7

Time Awake Since Boot: 45000 seconds
Time Since Wake:       3700 seconds

System Integrity Protection: disabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x00000a070284287d
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [0]

VM Regions Near 0xa070284287d:
    __LINKEDIT             00000001098b3000-00000001098c9000 [   88K] r--/rwx SM=COW  /usr/lib/dyld
--> 
    MALLOC_TINY            00007fb9abc00000-00007fb9abd00000 [ 1024K] rw-/rwx SM=PRV  

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   mutt                          	0x0000000100f2b340 url_free + 39
1   mutt                          	0x0000000100f49748 imap_parse_path + 406
2   mutt                          	0x0000000100f49517 imap_expand_path + 52
3   mutt                          	0x0000000100ef20ed _mutt_expand_path + 1348
4   mutt                          	0x0000000100ede70e parse_set + 1962
5   mutt                          	0x0000000100ee1c4a mutt_parse_rc_line + 276
6   mutt                          	0x0000000100ee3f15 source_rc + 633
7   mutt                          	0x0000000100ee3515 mutt_init + 2506
8   mutt                          	0x0000000100ee769b main + 2021
9   libdyld.dylib                 	0x00007fffbda75235 start + 1

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000007  rbx: 0x00000a070284286d  rcx: 0x000000000000000a  rdx: 0x0000000000000000
  rdi: 0x00007fff5ed500a8  rsi: 0x000000000000003a  rbp: 0x00007fff5ed50090  rsp: 0x00007fff5ed50060
   r8: 0x0000000000000013   r9: 0x00007fb9abe00b10  r10: 0x00000000fff80000  r11: 0xffffffba4d0afae0
  r12: 0x00007fff5ed50068  r13: 0x00007fff5ed505f0  r14: 0x00007fff5ed500a8  r15: 0x00007fff5ed500d8
  rip: 0x0000000100f2b340  rfl: 0x0000000000010202  cr2: 0x00000a070284287d
  
Logical CPU:     2
Error Code:      0x00000004
Trap Number:     14


Binary Images:
       0x100eab000 -        0x100f5cffb +mutt (0) <B2AB6E5B-402E-3F5F-B2F7-394B59F1260B> /opt/local2/*/mutt
       0x100f8b000 -        0x100fb9ffb +libidn.11.dylib (0) <66335A84-9B06-3B1C-ABA4-D16F5875AB72> /opt/local2/*/libidn.11.dylib
       0x100fc1000 -        0x100fd1ff3 +libz.1.dylib (0) <B1847624-7DA1-3A70-9A59-20BC265A3D99> /opt/local2/*/libz.1.dylib
       0x100fd5000 -        0x10101eff7 +libssl.1.0.0.dylib (0) <001C9013-B904-3D74-8296-ADD2A86ED8EF> /opt/local2/*/libssl.1.0.0.dylib
       0x10103d000 -        0x1011c0437 +libcrypto.1.0.0.dylib (0) <DBF789A4-7EA9-379E-BF57-D503578C53A8> /opt/local2/*/libcrypto.1.0.0.dylib
       0x10123f000 -        0x101279ffb +libncurses.6.dylib (0) <3787D6BD-F691-3A6C-A738-AB560DDE7458> /opt/local2/*/libncurses.6.dylib
       0x10128f000 -        0x101297ff7 +libintl.8.dylib (0) <1CB7824D-2B3E-30D1-B210-CCED90E63843> /opt/local2/*/libintl.8.dylib
       0x1012a2000 -        0x101396ff7 +libiconv.2.dylib (0) <0E649660-F8FE-37DF-898D-B709533B1D75> /opt/local2/*/libiconv.2.dylib
       0x10983e000 -        0x10987bdc7  dyld (433.5) <93EA668C-4BB2-3646-96E9-B406B81308D6> /usr/lib/dyld
    0x7fffa8268000 -     0x7fffa8701ff7  com.apple.CoreFoundation (6.9 - 1349.8) <09ED473E-5DE8-307F-B55C-16F6419236D5> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
    0x7fffbc28c000 -     0x7fffbc28dff3  libDiagnosticMessagesClient.dylib (102) <84A04D24-0E60-3810-A8C0-90A65E2DF61A> /usr/lib/libDiagnosticMessagesClient.dylib
    0x7fffbc4cb000 -     0x7fffbc4ccffb  libSystem.B.dylib (1238.60.2) <838E181D-C87F-3C8C-A6D9-FF80B2D05BEB> /usr/lib/libSystem.B.dylib
    0x7fffbc605000 -     0x7fffbc65bff7  libc++.1.dylib (307.5) <0B43BB5D-E6EB-3464-8DE9-B41AC8ED9D1C> /usr/lib/libc++.1.dylib
    0x7fffbc65c000 -     0x7fffbc685ff7  libc++abi.dylib (307.4) <BC271AD3-831B-362A-9DA7-E8C51F285FE4> /usr/lib/libc++abi.dylib
    0x7fffbcbc6000 -     0x7fffbcdebffb  libicucore.A.dylib (57166.0.1) <CCD2ED24-3071-383B-925D-8D763BB12A6F> /usr/lib/libicucore.A.dylib
    0x7fffbd17a000 -     0x7fffbd54c047  libobjc.A.dylib (709.1) <70614861-0340-32E2-85ED-FE65759CDFFA> /usr/lib/libobjc.A.dylib
    0x7fffbd979000 -     0x7fffbd98aff3  libz.1.dylib (67) <46E3FFA2-4328-327A-8D34-A03E20BFFB8E> /usr/lib/libz.1.dylib
    0x7fffbd999000 -     0x7fffbd99dff7  libcache.dylib (79) <093A4DAB-8385-3D47-A350-E20CB7CCF7BF> /usr/lib/system/libcache.dylib
    0x7fffbd99e000 -     0x7fffbd9a8fff  libcommonCrypto.dylib (60092.50.5) <8A64D1B0-C70E-385C-92F0-E669079FDA90> /usr/lib/system/libcommonCrypto.dylib
    0x7fffbd9a9000 -     0x7fffbd9b0fff  libcompiler_rt.dylib (62) <55D47421-772A-32AB-B529-1A46C2F43B4D> /usr/lib/system/libcompiler_rt.dylib
    0x7fffbd9b1000 -     0x7fffbd9b9fff  libcopyfile.dylib (138) <819BEA3C-DF11-3E3D-A1A1-5A51C5BF1961> /usr/lib/system/libcopyfile.dylib
    0x7fffbd9ba000 -     0x7fffbda3dfdf  libcorecrypto.dylib (442.50.19) <65D7165E-2E71-335D-A2D6-33F78E2DF0C1> /usr/lib/system/libcorecrypto.dylib
    0x7fffbda3e000 -     0x7fffbda6ffff  libdispatch.dylib (703.50.37) <6582BAD6-ED27-3B30-B620-90B1C5A4AE3C> /usr/lib/system/libdispatch.dylib
    0x7fffbda70000 -     0x7fffbda75ffb  libdyld.dylib (433.5) <83A214CD-AAE5-3ECD-85AC-B5E44FA4F37D> /usr/lib/system/libdyld.dylib
    0x7fffbda76000 -     0x7fffbda76ffb  libkeymgr.dylib (28) <7AA011A9-DC21-3488-BF73-3B5B14D1FDD6> /usr/lib/system/libkeymgr.dylib
    0x7fffbda84000 -     0x7fffbda84fff  liblaunch.dylib (972.70.1) <B856ABD2-896E-3DE0-B2C8-146A6AF8E2A7> /usr/lib/system/liblaunch.dylib
    0x7fffbda85000 -     0x7fffbda8aff3  libmacho.dylib (898) <17D5D855-F6C3-3B04-B680-E9BF02EF8AED> /usr/lib/system/libmacho.dylib
    0x7fffbda8b000 -     0x7fffbda8dff3  libquarantine.dylib (85.50.1) <12448CC2-378E-35F3-BE33-9DC395A5B970> /usr/lib/system/libquarantine.dylib
    0x7fffbda8e000 -     0x7fffbda8fffb  libremovefile.dylib (45) <38D4CB9C-10CD-30D3-8B7B-A515EC75FE85> /usr/lib/system/libremovefile.dylib
    0x7fffbda90000 -     0x7fffbdaa8ff7  libsystem_asl.dylib (349.50.5) <096E4228-3B7C-30A6-8B13-EC909A64499A> /usr/lib/system/libsystem_asl.dylib
    0x7fffbdaa9000 -     0x7fffbdaa9ff7  libsystem_blocks.dylib (67) <10DC5404-73AB-35B3-A277-A8AFECB476EB> /usr/lib/system/libsystem_blocks.dylib
    0x7fffbdaaa000 -     0x7fffbdb37fef  libsystem_c.dylib (1158.50.2) <E5AE5244-7D0C-36AC-8BB6-C7AE7EA52A4B> /usr/lib/system/libsystem_c.dylib
    0x7fffbdb38000 -     0x7fffbdb3bffb  libsystem_configuration.dylib (888.60.2) <BECC01A2-CA8D-31E6-BCDF-D452965FA976> /usr/lib/system/libsystem_configuration.dylib
    0x7fffbdb3c000 -     0x7fffbdb3ffff  libsystem_coreservices.dylib (41.4) <7D26DE79-B424-3450-85E1-F7FAB32714AB> /usr/lib/system/libsystem_coreservices.dylib
    0x7fffbdb40000 -     0x7fffbdb58fff  libsystem_coretls.dylib (121.50.4) <EC6FCF07-DCFB-3A03-9CC9-6DD3709974C6> /usr/lib/system/libsystem_coretls.dylib
    0x7fffbdb59000 -     0x7fffbdb5ffff  libsystem_dnssd.dylib (765.50.9) <CC960215-0B1B-3822-A13A-3DDE96FA796F> /usr/lib/system/libsystem_dnssd.dylib
    0x7fffbdb60000 -     0x7fffbdb89ff7  libsystem_info.dylib (503.50.4) <611DB84C-BF70-3F92-8702-B9F28A900920> /usr/lib/system/libsystem_info.dylib
    0x7fffbdb8a000 -     0x7fffbdbacff7  libsystem_kernel.dylib (3789.71.6) <5F3C87FD-A59C-390A-B0F0-26C831200FBD> /usr/lib/system/libsystem_kernel.dylib
    0x7fffbdbad000 -     0x7fffbdbf4fe7  libsystem_m.dylib (3121.6) <86D499B5-BBDC-3D3B-8A4E-97AE8E6672A4> /usr/lib/system/libsystem_m.dylib
    0x7fffbdbf5000 -     0x7fffbdc13ff7  libsystem_malloc.dylib (116.50.8) <A3D15F17-99A6-3367-8C7E-4280E8619C95> /usr/lib/system/libsystem_malloc.dylib
    0x7fffbdc14000 -     0x7fffbdc6dffb  libsystem_network.dylib (856.60.1) <369D0221-56CA-3C3E-9EDE-94B41CAE77B7> /usr/lib/system/libsystem_network.dylib
    0x7fffbdc6e000 -     0x7fffbdc77ff3  libsystem_networkextension.dylib (563.60.2) <B021F2B3-8A75-3633-ABB0-FC012B8E9B0C> /usr/lib/system/libsystem_networkextension.dylib
    0x7fffbdc78000 -     0x7fffbdc81ff3  libsystem_notify.dylib (165.20.1) <B8160190-A069-3B3A-BDF6-2AA408221FAE> /usr/lib/system/libsystem_notify.dylib
    0x7fffbdc82000 -     0x7fffbdc8afe7  libsystem_platform.dylib (126.50.8) <897462FD-B318-321B-A554-E61982630F7E> /usr/lib/system/libsystem_platform.dylib
    0x7fffbdc8b000 -     0x7fffbdc95ff7  libsystem_pthread.dylib (218.60.3) <B8FB5E20-3295-39E2-B5EB-B464D1D4B104> /usr/lib/system/libsystem_pthread.dylib
    0x7fffbdc96000 -     0x7fffbdc99ff7  libsystem_sandbox.dylib (592.70.2) <19320A42-2E3B-361B-BBDA-2F5F2E87B100> /usr/lib/system/libsystem_sandbox.dylib
    0x7fffbdc9a000 -     0x7fffbdc9bff3  libsystem_secinit.dylib (24.50.4) <F78B847B-3565-3E4B-98A6-F7AD40392E2D> /usr/lib/system/libsystem_secinit.dylib
    0x7fffbdc9c000 -     0x7fffbdca3ffb  libsystem_symptoms.dylib (532.50.47) <3390E07C-C1CE-348F-ADBD-2C5440B45EAA> /usr/lib/system/libsystem_symptoms.dylib
    0x7fffbdca4000 -     0x7fffbdcb7ff7  libsystem_trace.dylib (518.70.1) <AC63A7FE-50D9-3A30-96E6-F6B7FF16E465> /usr/lib/system/libsystem_trace.dylib
    0x7fffbdcb8000 -     0x7fffbdcbdffb  libunwind.dylib (35.3) <3D50D8A8-C460-334D-A519-2DA841102C6B> /usr/lib/system/libunwind.dylib
    0x7fffbdcbe000 -     0x7fffbdce7ff7  libxpc.dylib (972.70.1) <BF896DF0-D8E9-31A8-A4B3-01120BFEEE52> /usr/lib/system/libxpc.dylib

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 52704
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=133.4M resident=0K(0%) swapped_out_or_unallocated=133.4M(100%)
Writable regions: Total=28.5M written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=28.5M(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Activity Tracing                   256K        2 
Kernel Alloc Once                    8K        2 
MALLOC                            20.2M       10 
MALLOC guard page                   16K        4 
STACK GUARD                       56.0M        2 
Stack                             8192K        2 
__DATA                            3208K       56 
__LINKEDIT                       114.8M       11 
__TEXT                            18.6M       52 
__UNICODE                          556K        2 
shared memory                       12K        4 
===========                     =======  ======= 
TOTAL                            221.5M      136 
```
crash dump 2:
```
Process:               neomutt [60710]
Path:                  /opt/local2/*/neomutt
Identifier:            neomutt
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        bash [60528]
Responsible:           neomutt [60710]
User ID:               25161

Date/Time:             2017-11-03 22:35:59.653 +0100
OS Version:            Mac OS X 10.12.6 (16G1036)
Report Version:        12
Anonymous UUID:        69AD0AE9-9689-39A0-1F78-564B65702D8B

Sleep/Wake UUID:       44902E2C-5BF8-4A27-85C5-2C59258C56E7

Time Awake Since Boot: 45000 seconds
Time Since Wake:       3700 seconds

System Integrity Protection: disabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
abort() called
*** error for object 0x7fffbdaaa610: pointer being freed was not allocated
 

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	0x00007fffbdba3d42 __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fffbdc91457 pthread_kill + 90
2   libsystem_c.dylib             	0x00007fffbdb09420 abort + 129
3   libsystem_malloc.dylib        	0x00007fffbdbf8fe7 free + 530
4   mutt                          	0x000000010299049e safe_free + 27
5   mutt                          	0x000000010298a34c url_free + 51
6   mutt                          	0x00000001029a8748 imap_parse_path + 406
7   mutt                          	0x00000001029a8517 imap_expand_path + 52
8   mutt                          	0x00000001029510ed _mutt_expand_path + 1348
9   mutt                          	0x000000010293d70e parse_set + 1962
10  mutt                          	0x0000000102940c4a mutt_parse_rc_line + 276
11  mutt                          	0x0000000102942f15 source_rc + 633
12  mutt                          	0x0000000102942515 mutt_init + 2506
13  mutt                          	0x000000010294669b main + 2021
14  libdyld.dylib                 	0x00007fffbda75235 start + 1

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000000000006  rcx: 0x00007fff5d2f0f78  rdx: 0x0000000000000000
  rdi: 0x0000000000000307  rsi: 0x0000000000000006  rbp: 0x00007fff5d2f0fa0  rsp: 0x00007fff5d2f0f78
   r8: 0x0000000000000000   r9: 0x0000000000000000  r10: 0x0000000008000000  r11: 0x0000000000000206
  r12: 0x00007fff5d2f1058  r13: 0x0000000000000001  r14: 0x00007fffc69923c0  r15: 0x0000000000000000
  rip: 0x00007fffbdba3d42  rfl: 0x0000000000000206  cr2: 0x00007fffc6974128
  
Logical CPU:     0
Error Code:      0x02000148
Trap Number:     133


Binary Images:
       0x10290a000 -        0x1029bbffb +mutt (0) <B2AB6E5B-402E-3F5F-B2F7-394B59F1260B> /opt/local2/*/mutt
       0x1029ee000 -        0x102a1cffb +libidn.11.dylib (0) <66335A84-9B06-3B1C-ABA4-D16F5875AB72> /opt/local2/*/libidn.11.dylib
       0x102a23000 -        0x102a33ff3 +libz.1.dylib (0) <B1847624-7DA1-3A70-9A59-20BC265A3D99> /opt/local2/*/libz.1.dylib
       0x102a3e000 -        0x102a87ff7 +libssl.1.0.0.dylib (0) <001C9013-B904-3D74-8296-ADD2A86ED8EF> /opt/local2/*/libssl.1.0.0.dylib
       0x102aaa000 -        0x102c2d437 +libcrypto.1.0.0.dylib (0) <DBF789A4-7EA9-379E-BF57-D503578C53A8> /opt/local2/*/libcrypto.1.0.0.dylib
       0x102ca9000 -        0x102ce3ffb +libncurses.6.dylib (0) <3787D6BD-F691-3A6C-A738-AB560DDE7458> /opt/local2/*/libncurses.6.dylib
       0x102cf6000 -        0x102cfeff7 +libintl.8.dylib (0) <1CB7824D-2B3E-30D1-B210-CCED90E63843> /opt/local2/*/libintl.8.dylib
       0x102d09000 -        0x102dfdff7 +libiconv.2.dylib (0) <0E649660-F8FE-37DF-898D-B709533B1D75> /opt/local2/*/libiconv.2.dylib
       0x111537000 -        0x111574dc7  dyld (433.5) <93EA668C-4BB2-3646-96E9-B406B81308D6> /usr/lib/dyld
    0x7fffa8268000 -     0x7fffa8701ff7  com.apple.CoreFoundation (6.9 - 1349.8) <09ED473E-5DE8-307F-B55C-16F6419236D5> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
    0x7fffbc28c000 -     0x7fffbc28dff3  libDiagnosticMessagesClient.dylib (102) <84A04D24-0E60-3810-A8C0-90A65E2DF61A> /usr/lib/libDiagnosticMessagesClient.dylib
    0x7fffbc4cb000 -     0x7fffbc4ccffb  libSystem.B.dylib (1238.60.2) <838E181D-C87F-3C8C-A6D9-FF80B2D05BEB> /usr/lib/libSystem.B.dylib
    0x7fffbc605000 -     0x7fffbc65bff7  libc++.1.dylib (307.5) <0B43BB5D-E6EB-3464-8DE9-B41AC8ED9D1C> /usr/lib/libc++.1.dylib
    0x7fffbc65c000 -     0x7fffbc685ff7  libc++abi.dylib (307.4) <BC271AD3-831B-362A-9DA7-E8C51F285FE4> /usr/lib/libc++abi.dylib
    0x7fffbcbc6000 -     0x7fffbcdebffb  libicucore.A.dylib (57166.0.1) <CCD2ED24-3071-383B-925D-8D763BB12A6F> /usr/lib/libicucore.A.dylib
    0x7fffbd17a000 -     0x7fffbd54c047  libobjc.A.dylib (709.1) <70614861-0340-32E2-85ED-FE65759CDFFA> /usr/lib/libobjc.A.dylib
    0x7fffbd979000 -     0x7fffbd98aff3  libz.1.dylib (67) <46E3FFA2-4328-327A-8D34-A03E20BFFB8E> /usr/lib/libz.1.dylib
    0x7fffbd999000 -     0x7fffbd99dff7  libcache.dylib (79) <093A4DAB-8385-3D47-A350-E20CB7CCF7BF> /usr/lib/system/libcache.dylib
    0x7fffbd99e000 -     0x7fffbd9a8fff  libcommonCrypto.dylib (60092.50.5) <8A64D1B0-C70E-385C-92F0-E669079FDA90> /usr/lib/system/libcommonCrypto.dylib
    0x7fffbd9a9000 -     0x7fffbd9b0fff  libcompiler_rt.dylib (62) <55D47421-772A-32AB-B529-1A46C2F43B4D> /usr/lib/system/libcompiler_rt.dylib
    0x7fffbd9b1000 -     0x7fffbd9b9fff  libcopyfile.dylib (138) <819BEA3C-DF11-3E3D-A1A1-5A51C5BF1961> /usr/lib/system/libcopyfile.dylib
    0x7fffbd9ba000 -     0x7fffbda3dfdf  libcorecrypto.dylib (442.50.19) <65D7165E-2E71-335D-A2D6-33F78E2DF0C1> /usr/lib/system/libcorecrypto.dylib
    0x7fffbda3e000 -     0x7fffbda6ffff  libdispatch.dylib (703.50.37) <6582BAD6-ED27-3B30-B620-90B1C5A4AE3C> /usr/lib/system/libdispatch.dylib
    0x7fffbda70000 -     0x7fffbda75ffb  libdyld.dylib (433.5) <83A214CD-AAE5-3ECD-85AC-B5E44FA4F37D> /usr/lib/system/libdyld.dylib
    0x7fffbda76000 -     0x7fffbda76ffb  libkeymgr.dylib (28) <7AA011A9-DC21-3488-BF73-3B5B14D1FDD6> /usr/lib/system/libkeymgr.dylib
    0x7fffbda84000 -     0x7fffbda84fff  liblaunch.dylib (972.70.1) <B856ABD2-896E-3DE0-B2C8-146A6AF8E2A7> /usr/lib/system/liblaunch.dylib
    0x7fffbda85000 -     0x7fffbda8aff3  libmacho.dylib (898) <17D5D855-F6C3-3B04-B680-E9BF02EF8AED> /usr/lib/system/libmacho.dylib
    0x7fffbda8b000 -     0x7fffbda8dff3  libquarantine.dylib (85.50.1) <12448CC2-378E-35F3-BE33-9DC395A5B970> /usr/lib/system/libquarantine.dylib
    0x7fffbda8e000 -     0x7fffbda8fffb  libremovefile.dylib (45) <38D4CB9C-10CD-30D3-8B7B-A515EC75FE85> /usr/lib/system/libremovefile.dylib
    0x7fffbda90000 -     0x7fffbdaa8ff7  libsystem_asl.dylib (349.50.5) <096E4228-3B7C-30A6-8B13-EC909A64499A> /usr/lib/system/libsystem_asl.dylib
    0x7fffbdaa9000 -     0x7fffbdaa9ff7  libsystem_blocks.dylib (67) <10DC5404-73AB-35B3-A277-A8AFECB476EB> /usr/lib/system/libsystem_blocks.dylib
    0x7fffbdaaa000 -     0x7fffbdb37fef  libsystem_c.dylib (1158.50.2) <E5AE5244-7D0C-36AC-8BB6-C7AE7EA52A4B> /usr/lib/system/libsystem_c.dylib
    0x7fffbdb38000 -     0x7fffbdb3bffb  libsystem_configuration.dylib (888.60.2) <BECC01A2-CA8D-31E6-BCDF-D452965FA976> /usr/lib/system/libsystem_configuration.dylib
    0x7fffbdb3c000 -     0x7fffbdb3ffff  libsystem_coreservices.dylib (41.4) <7D26DE79-B424-3450-85E1-F7FAB32714AB> /usr/lib/system/libsystem_coreservices.dylib
    0x7fffbdb40000 -     0x7fffbdb58fff  libsystem_coretls.dylib (121.50.4) <EC6FCF07-DCFB-3A03-9CC9-6DD3709974C6> /usr/lib/system/libsystem_coretls.dylib
    0x7fffbdb59000 -     0x7fffbdb5ffff  libsystem_dnssd.dylib (765.50.9) <CC960215-0B1B-3822-A13A-3DDE96FA796F> /usr/lib/system/libsystem_dnssd.dylib
    0x7fffbdb60000 -     0x7fffbdb89ff7  libsystem_info.dylib (503.50.4) <611DB84C-BF70-3F92-8702-B9F28A900920> /usr/lib/system/libsystem_info.dylib
    0x7fffbdb8a000 -     0x7fffbdbacff7  libsystem_kernel.dylib (3789.71.6) <5F3C87FD-A59C-390A-B0F0-26C831200FBD> /usr/lib/system/libsystem_kernel.dylib
    0x7fffbdbad000 -     0x7fffbdbf4fe7  libsystem_m.dylib (3121.6) <86D499B5-BBDC-3D3B-8A4E-97AE8E6672A4> /usr/lib/system/libsystem_m.dylib
    0x7fffbdbf5000 -     0x7fffbdc13ff7  libsystem_malloc.dylib (116.50.8) <A3D15F17-99A6-3367-8C7E-4280E8619C95> /usr/lib/system/libsystem_malloc.dylib
    0x7fffbdc14000 -     0x7fffbdc6dffb  libsystem_network.dylib (856.60.1) <369D0221-56CA-3C3E-9EDE-94B41CAE77B7> /usr/lib/system/libsystem_network.dylib
    0x7fffbdc6e000 -     0x7fffbdc77ff3  libsystem_networkextension.dylib (563.60.2) <B021F2B3-8A75-3633-ABB0-FC012B8E9B0C> /usr/lib/system/libsystem_networkextension.dylib
    0x7fffbdc78000 -     0x7fffbdc81ff3  libsystem_notify.dylib (165.20.1) <B8160190-A069-3B3A-BDF6-2AA408221FAE> /usr/lib/system/libsystem_notify.dylib
    0x7fffbdc82000 -     0x7fffbdc8afe7  libsystem_platform.dylib (126.50.8) <897462FD-B318-321B-A554-E61982630F7E> /usr/lib/system/libsystem_platform.dylib
    0x7fffbdc8b000 -     0x7fffbdc95ff7  libsystem_pthread.dylib (218.60.3) <B8FB5E20-3295-39E2-B5EB-B464D1D4B104> /usr/lib/system/libsystem_pthread.dylib
    0x7fffbdc96000 -     0x7fffbdc99ff7  libsystem_sandbox.dylib (592.70.2) <19320A42-2E3B-361B-BBDA-2F5F2E87B100> /usr/lib/system/libsystem_sandbox.dylib
    0x7fffbdc9a000 -     0x7fffbdc9bff3  libsystem_secinit.dylib (24.50.4) <F78B847B-3565-3E4B-98A6-F7AD40392E2D> /usr/lib/system/libsystem_secinit.dylib
    0x7fffbdc9c000 -     0x7fffbdca3ffb  libsystem_symptoms.dylib (532.50.47) <3390E07C-C1CE-348F-ADBD-2C5440B45EAA> /usr/lib/system/libsystem_symptoms.dylib
    0x7fffbdca4000 -     0x7fffbdcb7ff7  libsystem_trace.dylib (518.70.1) <AC63A7FE-50D9-3A30-96E6-F6B7FF16E465> /usr/lib/system/libsystem_trace.dylib
    0x7fffbdcb8000 -     0x7fffbdcbdffb  libunwind.dylib (35.3) <3D50D8A8-C460-334D-A519-2DA841102C6B> /usr/lib/system/libunwind.dylib
    0x7fffbdcbe000 -     0x7fffbdce7ff7  libxpc.dylib (972.70.1) <BF896DF0-D8E9-31A8-A4B3-01120BFEEE52> /usr/lib/system/libxpc.dylib

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 52704
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=133.4M resident=0K(0%) swapped_out_or_unallocated=133.4M(100%)
Writable regions: Total=36.5M written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=36.5M(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Activity Tracing                   256K        2 
Kernel Alloc Once                    8K        2 
MALLOC                            28.2M       11 
MALLOC guard page                   16K        4 
STACK GUARD                       56.0M        2 
Stack                             8192K        2 
VM_ALLOCATE                          4K        2 
__DATA                            3208K       56 
__LINKEDIT                       114.8M       11 
__TEXT                            18.6M       52 
__UNICODE                          556K        2 
shared memory                       12K        4 
===========                     =======  ======= 
TOTAL                            229.5M      138 
```

I am having trouble building this branch:
```
$ ./configure.autosetup --prefix=/opt/local --with-idn=/opt/local --with-ncurses=/opt/local --with-nls=/opt/local --with-ssl=/opt/local
...
Summary of build options:

  Version:           20171027
  Host OS:           darwin17.2.0
  Install prefix:    /opt/local
  Compiler:           cc
  CFlags:            -g -O2 -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/opt/local/include
  LDFlags:           -L/opt/local/lib
  Libs:              -lidn -lz -lssl -lcrypto -lncursesw -lintl -liconv
  Header cache libs:

  GPGME:             no
  PGP:               yes
  SMIME:             yes
  Notmuch:           no
  Header Cache(s):
  Lua:               no

$ make
...
cc -o neomutt mutt_account.o addrbook.o alias.o attach.o bcache.o body.o browser.o buffy.o charset.o color.o commands.o complete.o compose.o compress.o conststrings.o copy.o curs_lib.o curs_main.o edit.o editmsg.o enter.o envelope.o filter.o flags.o from.o group.o handler.o hdrline.o headers.o help.o history.o hook.o init.o keymap.o main.o mbox.o mbyte.o menu.o mh.o muttlib.o mutt_idna.o mutt_socket.o mutt_tags.o mx.o newsrc.o nntp.o pager.o parameter.o parse.o pattern.o pop.o pop_auth.o pop_lib.o postpone.o query.o recvattach.o recvcmd.o rfc1524.o rfc2047.o rfc2231.o rfc3676.o rfc822.o safe_asprintf.o score.o send.o sendlib.o sidebar.o signal.o smtp.o sort.o state.o status.o system.o thread.o url.o version.o resize.o libmutt.a libncrypt.a libimap.a libconn.a -L/opt/local/lib  -lidn -lz -lssl -lcrypto -lncursesw -lintl -liconv
Undefined symbols for architecture x86_64:
  "_CertificateFile", referenced from:
      _MuttVars in init.o
      _mutt_ssl_starttls in libconn.a(ssl.o)
      _ssl_load_certificates in libconn.a(ssl.o)
      _ssl_socket_open in libconn.a(ssl.o)
      _ssl_verify_callback in libconn.a(ssl.o)
      _interactive_check_cert in libconn.a(ssl.o)
  "_ConnectTimeout", referenced from:
      _MuttVars in init.o
      _raw_socket_open in libconn.a(socket.o)
  "_EntropyFile", referenced from:
      _MuttVars in init.o
      _ssl_init in libconn.a(ssl.o)
  "_Preconnect", referenced from:
      _MuttVars in init.o
      _mutt_socket_open in libconn.a(socket.o)
  "_SslCiphers", referenced from:
      _MuttVars in init.o
      _mutt_ssl_starttls in libconn.a(ssl.o)
      _ssl_socket_open in libconn.a(ssl.o)
  "_SslClientCert", referenced from:
      _MuttVars in init.o
      _ssl_get_client_cert in libconn.a(ssl.o)
  "_Tunnel", referenced from:
      _MuttVars in init.o
      _mutt_conn_find in mutt_socket.o
      _tunnel_socket_open in libconn.a(tunnel.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [neomutt] Error 1
```

Is there something I'm forgetting or doing wrong?

--- a/url.c
+++ b/url.c
@@ -152,18 +152,18 @@ int url_parse(struct Url *u, char *src)
 {
   char *t = NULL, *p = NULL;
 
-  u->scheme = url_check_scheme(src);
-  if (u->scheme == U_UNKNOWN)
-    return -1;
-
-  src = strchr(src, ':') + 1;
-
   u->user = NULL;
   u->pass = NULL;
   u->host = NULL;
   u->port = 0;
   u->path = NULL;
   STAILQ_INIT(&u->query_strings);
+
+  u->scheme = url_check_scheme(src);
+  if (u->scheme == U_UNKNOWN)
+    return -1;
+
+  src = strchr(src, ':') + 1;
 
   if (strncmp(src, "//", 2) != 0)
   {

